In [1]:
import pandas as pd

In [2]:
state_df = pd.read_csv(r'C:\Users\billy\Documents\NSS_projects\open-secrets-skittles\Data\states_districts_df.csv')

In [3]:
CRPICO_Result = (
    state_df
    .groupby('CRPICO')['Result']
    .value_counts(normalize=True)
    .reset_index()
)
# I needed to compare the columns comprising the Incumbent, Challenger, and Open variables and result of the race.
# I reset it to create a pandas dataframe. 

In [4]:
CRPICO_Result

CRPICO Result  proportion
0       C      L    0.533058
1       C           0.450413
2       C      W    0.016529
3       I      W    0.961140
4       I      L    0.031088
5       I      w    0.005181
6       I           0.002591
7       O      L    0.394737
8       O      W    0.309211
9       O           0.282895
10      O      l    0.006579
11      O      w    0.006579

In [5]:
Incumbent_proportions = (
    CRPICO_Result[CRPICO_Result['CRPICO'] == 'I']
    .sort_values(by='proportion', ascending=False)
    .replace({'w':'W'})
)
# I subsetted the dataframe by I variable which represents the Incumbent and sorted by proportions.
# I renamed the lowercase w variables with uppercase in order to sum those value, which I did in the next cell.

In [6]:
Incumbent_proportions

CRPICO Result  proportion
3      I      W    0.961140
4      I      L    0.031088
5      I      W    0.005181
6      I           0.002591

In [7]:
Incumbent_proportions[Incumbent_proportions['Result'] == 'W']['proportion'].sum()
# I summed the W variables.

0.966321243523316

*The incumbent won a race 96.6% of the time in the 2020 congressional elections.*